app/src/main/java/one/mixin/android/job/SendTranscriptJob.kt

```diff
@@ -0,0 +1,110 @@
+            jobManager.start()
+        }
+        val conversation = conversationDao.findConversationById(message.conversationId)
+        if (conversation != null) {
+            val stringBuffer = StringBuffer()
+            transcriptMessages.filter { it.isText() || it.isPost() || it.isData() || it.isContact() }.forEach { transcript ->
+                if (transcript.isData()) {
+                    transcript.mediaName
+                } else if (transcript.isContact()) {
+                    transcript.sharedUserId?.let { userId -> userDao.findUser(userId) }?.fullName
+                } else {
+                    transcript.content
+                }?.joinWhiteSpace()?.let {
+                    stringBuffer.append(it)
+                }
+            }
+            MessageFts4Helper.insertMessageFts4(message.id, stringBuffer.toString())
+            messageDao.insert(message)
+            transcriptMessages.forEach { transcript ->
+                if (transcript.isAttachment()) {
+                    val mediaUrl = transcript.mediaUrl
+                    if (mediaUrl == null) {
+                        transcript.mediaUrl = null
+                        transcript.mediaStatus = MediaStatus.DONE.name
+                    } else {
+                        val file = File(Uri.parse(mediaUrl).path!!)
+                        if (file.exists()) {
+                            val outFile = MixinApplication.appContext.getTranscriptFile(
+                                transcript.messageId,
+                                file.name.getExtensionName().notNullWithElse({ ".$it" }, ""),
+                            )
+                            if (!outFile.exists() || outFile.length() <= 0) {
+                                file.copy(outFile)
+                            }
+                            transcript.mediaUrl = outFile.toUri().toString()
+                            transcript.mediaStatus = MediaStatus.CANCELED.name
+                        } else {
+                            transcript.mediaUrl = null
+                            transcript.mediaStatus = MediaStatus.DONE.name
+                        }
+                    }
+                }
+            }
+            transcriptMessageDao.insertList(transcriptMessages)
+        } else {
+            Bugsnag.notify(Throwable("Insert failed, no conversation exist"))
+        }
+    }
+
+    override fun cancel() {
+    }
+
+    override fun onRun() {
+        val transcripts = mutableSetOf<TranscriptMessage>()
+        getTranscripts(message.id, transcripts)
+
+        if (transcripts.any { t -> t.isAttachment() }) {
+            val mediaSize = transcripts.sumOf { t -> t.mediaSize ?: 0 }
+            messageDao.updateMediaSize(mediaSize, message.id)
+            transcripts.filter { t ->
+                t.isAttachment()
+            }.forEach { t ->
+                jobManager.addJob(SendTranscriptAttachmentMessageJob(t, message.isPlain(), message.id))
+            }
```

Comment:
not handle non-attachment messages?

Reply:
The last step of all attachment upload job is to package the data to send